structs.go

```diff
@@ -79,6 +83,11 @@ func New(s interface{}) *Struct {
 // Note that only exported fields of a struct can be accessed, non exported
 // fields will be neglected.
 func (s *Struct) Map() map[string]interface{} {
+	val := s.value.Interface()
+	if mapper, ok := val.(Mapper); ok {
```

Comment:
Val never implements a Mapper, because when calling `structs.New` the pointer is removed. That would need to be changed if we wanted to make this suggestion work